src/ray/protobuf/autoscaler.proto

```diff
@@ -0,0 +1,172 @@
+  // The group is not allowed to deploy more than one bundle on a node.
+  STRICT_SPREAD = 3;
+}
+
+enum SourceType {
+  // A task that (assuming no deadlock can eventually be scheduled on the cluster.
+  TASK = 0;
+
+  // A actor that (assuming no deadlock can eventually be scheduled on the cluster.
+  ACTOR = 1;
+
+  // A placement group that (assuming no deadlock can eventually be scheduled on the cluster.
+  PLACEMENT_GROUP = 2;
+
+  // Represents a job's call to
+  // `ray.autoscaler.sdk.request_resources()`. This type requests
+  // resources to be available on the cluster but won't actually
+  // utilize them.
+  SDK_RESOURCE_REQUEST = 3;
+}
+
+// A workload (task/actor/placement group) unfulfilled by the the cluster's internal scheduler(s).
+message QueuedResources {
+  // The number of resources of this quantity. This should be
+  // functionally equivalent to setting quantity=1 and repeating the
+  // field many times.
+  uint32 quantity = 1;
+
+  // Whether the queued resources can eventually be scheduled on the cluster
+  // (neglecting deadlock).
+  bool is_feasible = 2;
+
+  // The source of this queued resource.
+  SourceType type = 3;
+
+  // If the queued resource represents an actor or task, there is
+  // 1 entry in `bundles`. In a placement group, each bundle
+  // receives its own entry.
+  repeated ResourceBundle bundles = 4;
+
+  // The placement strategy of bundles within this queued resources.
+  PlacementStrategy placement_strategy = 5;
+}
+
+message GetAllQueuedResourcesRequest {
+}
+
+message GetAllQueuedResourcesResponse {
+  // The set of all queued resources.
+  repeated QueuedResources queued_resources = 1;
+}
+
+message NodeInfo {
+  // State of a node.
+  enum NodeState {
+    // The node has not joined the cluster yet.
+    PENDING = 0;
+
+    // The node has joined the cluser and is healthy.
+    ALIVE = 1;
+
+    // The node is marked for removal, but it may not be safe to remove it yet.
+    DRAINING = 2;
+
+    // The node is no longer active in the cluster it is safe to remove it.
+    DEAD = 3;
+  }
+
+  // The state of the node.
+  NodeState node_state = 1;
+
+  // TODO: Ideally we unify GCS's node id with the autoscaler's here.
+  // The ID of node.
+  bytes node_id = 2;
+
+  // The address of the node. This information is primarily
+  // intended for user-facing observability. The processes at this
+  // address don't provide a stable api.
+  string node_manager_address = 3;
+
+  // The available fixed point resources (granularity 1/10000) of this node.
+  map<string, int64> resources_available = 4;
+  // The total fixed point resources (granularity 1/10000) of this node.
+  map<string, int64> resources_total = 5;
+
+  // The underlying host type (e.g. instance type or machine type).
+  string host_type = 6;
+}
+
+message GetAllNodeStatusesRequest {
+}
+
+message GetAllNodeStatusesResponse {
+  // The set of all nodes that ever were or ever will be in the cluster.
```

Comment:
well..."will be" that we know of

Comment is clear already,
but something like "Set of all past, present, and pending nodes" is more precise.